plugin/pkg/scheduler/api/types.go

```diff
@@ -100,3 +108,61 @@ type LabelPreference struct {
 	// If false, higher priority is given to nodes that do not have the label
 	Presence bool `json:"presence"`
 }
+
+// Holds the parameters used to communicate with the extender. If a verb is unspecified/empty,
+// it is assumed that the extender chose not to provide that extension.
+type ExtenderConfig struct {
+	// URLPrefix at which the extender is available
+	URLPrefix string `json:"urlPrefix"`
+	// Verb for the filter call, empty if not supported
+	FilterVerb string `json:"filterVerb,omitempty"`
+	// Verb for the prioritize call, empty if not supported
+	PrioritizeVerb string `json:"prioritizeVerb,omitempty"`
+	// The numeric multiplier for the node scores that the prioritize call generates.
+	// The weight should be a positive integer
+	Weight int `json:"weight,omitempty"`
+	// EnableHttps specifies whether https should be used to communicate with the extender
+	EnableHttps bool `json:"enableHttps,omitempty"`
+	// TLSConfig specifies the transport layer security config
+	TLSConfig *client.TLSClientConfig `json:"tlsConfig,omitempty"`
+	// HTTPTimeout specifies the timeout duration for a call to the extender. Filter timeout
+	// fails the scheduling of the pod. Prioritize timeout is ignored, k8s/other extenders
+	// priorities are used to select the node.
+	HTTPTimeout time.Duration `json:"httpTimeout,omitempty"`
+}
```

Comment:
Completely unrelated, but if we moved things like this to their own file we could have the docs be automatically generated/updated...

Not sure if we want to introduce that pattern into the project, as it hides the `type` away from the code author...   I wonder how hard it would be to get the munger to handle it in place...

No need to address this comment, just thinking outloud...